parsers/deathloopvaccine.py

```diff
@@ -0,0 +1,286 @@
+                # player is not AFK
+                afk = False
+                starprint(f'DeathLoopVaccine:  Player Not AFK: {line}')
+
+            # does this line contain a proof of life - melee
+            regexp = r'^You( try to)? (hit|slash|pierce|crush|claw|bite|sting|maul|gore|punch|kick|backstab|bash)'
+            m = re.match(regexp, trunc_line)
+            if m:
+                # player is not AFK
+                afk = False
+                starprint(f'DeathLoopVaccine:  Player Not AFK: {line}')
+
+            # if they are not AFK, then go ahead and purge any death messages from the list
+            if not afk:
+                self.reset()
+
+    def deathloop_response(self) -> None:
+        """
+        are we death looping?  if so, kill the process
+
+        Returns:
+            None:
+        """
+
+        # if the death_list contains more deaths than the limit, then trigger the process kill
+        if len(self._death_list) >= self.deathloop_deaths:
+
+            starprint('---------------------------------------------------')
+            starprint('DeathLoopVaccine - Killing all eqgame.exe processes')
+            starprint('---------------------------------------------------')
+            starprint('DeathLoopVaccine has detected deathloop symptoms:')
+            starprint(f'    {self.deathloop_deaths} deaths in less than {self.deathloop_seconds} seconds, with no player activity')
+
+            # show all the death messages
+            starprint('Death Messages:')
+            for line in self._death_list:
+                starprint('    ' + line)
+
+            # get the list of eqgame.exe process ID's, and show them
+            pid_list = get_eqgame_pid_list()
+            starprint(f'eqgame.exe process id list = {pid_list}')
+
+            # kill the eqgame.exe process / processes
+            for pid in pid_list:
+                starprint(f'Killing process [{pid}]')
+
+                # for testing the actual kill process using simulated player deaths, uncomment the following line
+                # self._kill_armed = True
+                if self._kill_armed:
+                    os.kill(pid, signal.SIGTERM)
+                else:
+                    starprint('(Note: Process Kill only simulated, since death(s) were simulated)')
+
+            # purge any death messages from the list
+            self.reset()
+
+
+#################################################################################################
+#
+# standalone functions
+#
+
+
+def get_eqgame_pid_list() -> list[int]:
+    """
+    get list of process ID's for eqgame.exe, using psutil module
+
+    Returns:
+        object: list of process ID's (in case multiple versions of eqgame.exe are somehow running)
+    """
+
+    pid_list = list()
+    for p in psutil.process_iter(['name']):
+        if p.info['name'] == 'eqgame.exe':
+            pid_list.append(p.pid)
+    return pid_list
+
+
+def starprint(line: str) -> None:
```

Comment:
We don't really need most print related stuff in nParse since there's no console

Reply:
The .exe could be built with a console window (the standalone DLV does that).  I've been running nparse using the python command line 'py nparse.py' and it provides a console window with lots of lovely info.

Reply:
moved it to helpers